appc/embedded.go

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package appc implements App Connectors. An AppConnector provides domain
+// oriented routing of traffic.
+package appc
+
+import (
+	"net/netip"
+	"slices"
+	"sync"
+
+	"golang.org/x/net/dns/dnsmessage"
+	"tailscale.com/types/logger"
+)
+
+/*
+ * TODO(raggi): the sniproxy servicing portions of this package will be moved
+ * into the sniproxy or deprecated at some point, when doing so is not
+ * disruptive. At that time EmbeddedAppConnector can be renamed to AppConnector.
+ */
+
+// RouteAdvertiser is an interface that allows the AppConnector to advertise
+// newly discovered routes that need to be served through the AppConnector.
+type RouteAdvertiser interface {
+	// AdvertiseRoute adds a new route advertisement if the route is not already
+	// being advertised.
+	AdvertiseRoute(netip.Prefix) error
+}
+
+// EmbeddedAppConnector is an implementation of an AppConnector that performs
```

Comment:
Might be worth adding a one-liner as to what an App Connector is: a proxy node for traffic to a bunch of things identified by DNS.

Reply:
That's added to the package doc at the top, which once we move the sniproxy files out should become the canonical one. I'll add more when we do that.